### ISSUE TITLE
[feature request] swarm: rolling update with no replicas

### ISSUE THREAD

**Issue by brunoqc** [docker/docker#29148]
**Description**

I'm able to do rolling update without any "Connection Refused" if I have 2 replicas but not if I have only one.

For people with small VPS, it would be useful to be able to do "rolling updates" with only one node and having the containers running only once.

**Steps to reproduce the issue:**
1. run `docker service create -p 8080:80 --name hello --update-delay 10s --stop-grace-period 10s dockercloud/hello-world`
2. run `siege http://127.0.0.1:8080`
3. run `docker service update --image dockercloud/hello-world:staging hello`
4. wait for the update to finish
5. stop 'siege'

**Describe the results you received:**
I see a lot of "Connection refused" and a non 100% availability.

**Describe the results you expected:**
no "Connection refused" and 100% availability.

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `docker version`:**

```
Client:
 Version:      1.14.0-dev
 API version:  1.26
 Go version:   go1.7.4
 Git commit:   1c96879
 Built:        Mon Dec  5 16:14:53 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.14.0-dev
 API version:  1.26 (minimum version 1.12)
 Go version:   go1.7.4
 Git commit:   1c96879
 Built:        Mon Dec  5 16:14:53 2016
 OS/Arch:      linux/amd64
 Experimental: false
```

**Output of `docker info`:**

```
Containers: 5
 Running: 1
 Paused: 0
 Stopped: 4
Images: 70
Server Version: 1.14.0-dev
Storage Driver: devicemapper
 Pool Name: docker-8:2-24772686-pool
 Pool Blocksize: 65.54 kB
 Base Device Size: 10.74 GB
 Backing Filesystem: xfs
 Data file: /dev/loop0
 Metadata file: /dev/loop1
 Data Space Used: 4.849 GB
 Data Space Total: 107.4 GB
 Data Space Available: 102.5 GB
 Metadata Space Used: 8.417 MB
 Metadata Space Total: 2.147 GB
 Metadata Space Available: 2.139 GB
 Thin Pool Minimum Free Space: 10.74 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: false
 Deferred Deletion Enabled: false
 Deferred Deleted Device Count: 0
 Data loop file: /var/lib/docker/devicemapper/devicemapper/data
 WARNING: Usage of loopback devices is strongly discouraged for production use. Use `--storage-opt dm.thinpooldev` to specify a custom block storage device.
 Metadata loop file: /var/lib/docker/devicemapper/devicemapper/metadata
 Library Version: 1.02.136 (2016-11-05)
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins: 
 Volume: local
 Network: bridge host macvlan null overlay
Swarm: active
 NodeID: 68a62tley9l6zk71erj1wnr6t
 Is Manager: true
 ClusterID: 0nad6052izxvfvwllyscr0s0v
 Managers: 1
 Nodes: 1
 Orchestration:
  Task History Retention Limit: 5
 Raft:
  Snapshot Interval: 10000
  Number of Old Snapshots to Retain: 0
  Heartbeat Tick: 1
  Election Tick: 3
 Dispatcher:
  Heartbeat Period: 5 seconds
 CA Configuration:
  Expiry Duration: 3 months
 Node Address: 10.1.10.94
 Manager Addresses:
  10.1.10.94:2377
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 9c21822f344b7894ec2e55ac4b6d4e86118907dd (expected: 03e5862ec0d8d3b3f750e19fca3ee367e13c090e)
runc version: N/A (expected: 51371867a01c467f08af739783b8beafc154c4d7)
init version: N/A (expected: 949e6facb77383876aeff8a6944dde66b3089574)
Kernel Version: 4.8.11-1-ARCH
Operating System: Arch Linux
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 7.801 GiB
Name: bruno
ID: FQPI:5DOA:ESFX:4DUW:A3RE:GQ6A:UDLN:HRLV:I5HU:UTO4:3HX2:CCP2
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Username: bbigras
Registry: https://index.docker.io/v1/
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
physical

**Comment by cpuguy83** [comment]
ping @stevvooe

**Comment by thaJeztah** [comment]
I wonder if this is easy to achieve, or leads to a lot of unwanted side effects. Basically, this would mean temporarily scaling the service to _two_ instances, and once the second instance is "up", remove the first instance.

However, there are many cases where starting a second instance can result in problems. Think of a database service; having multiple instances of a database write to the same storage is generally not supported, and can lead to data corruption.

Also, since this requires (at least temporary) _double_  the amount of resources on the host, which can either lead to (on a single node/single manager) the manager running out of resources (and thus loosing control over the swarm), or the second instance of the service not being able to reserve resources, thus not starting.

**Comment by stevvooe** [comment]
You need to have two instances to get zero downtime rolling updates.

There are valid update models that scale an extra instance but there are some considerations to make when doing so.

cc @aaronlehmann
